core/src/services/fs/backend.rs

```diff
@@ -566,10 +578,24 @@ impl Accessor for FsBackend {
             (p, None)
         };
 
-        let f = std::fs::OpenOptions::new()
-            .create(true)
-            .truncate(true)
-            .write(true)
+        let mut f = std::fs::OpenOptions::new();
+        f.create(true).write(true);
+
+        if op.append() {
+            f.append(true);
+
+            // If the target file exists, we should append to the end of it directly.
+            if Path::new(&target_path)
```

Comment:
How about do this check in `if let Some(atomic_write_dir) {}` block, so that we can avoid those cost if `atomic_write` not enabled.

Reply:
Make sense.